lib/active_resource/detailed_log_subscriber.rb

```diff
@@ -10,8 +21,18 @@ def request(event)
       info "Response:\n#{event.payload[:response].body}"
     end
 
-    def logger
-      ActiveResource::Base.logger
+    def warn_on_deprecated_header(event)
```

Comment:
maybe move this method to lib/shopify_api/connection under  `ShopifyAPI::RequestNotification.warn_on_deprecated_header` and then call `.warn_on_deprecated_header` in `.notify_about_request` 

then call again as part of `ActiveResource::DetailedLogSubscriber.log_request_response_details`?

Reply:
If I understand correctly, I think that would add coupling via direct method calls, and defeat the purpose of the pub/sub model currently being used, where `ShopifyAPI::Connection#request` ends up notifying subscribers of `request.active_resource_detailed` about the payload (req/res) it encountered.